cmd/cli/proxy_configdump.go

```diff
@@ -0,0 +1,122 @@
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+const dumpConfigDescription = `
+This command will dump the sidecar proxy configuration for the given pod.
+`
+
+const dumpConfigExample = `
+# Dump the proxy configuration for pod 'bookbuyer-5ccf77f46d-rc5mg' in the 'bookbuyer' namespace
+osm proxy dump-config bookbuyer-5ccf77f46d-rc5mg -n bookbuyer
+`
+
+type proxyDumpConfigCmd struct {
+	out        io.Writer
+	config     *rest.Config
+	clientSet  kubernetes.Interface
+	namespace  string
+	pod        string
+	localPort  uint16
+	sigintChan chan os.Signal
+}
+
+func newProxyDumpConfig(config *action.Configuration, out io.Writer) *cobra.Command {
+	dumpConfigCmd := &proxyDumpConfigCmd{
+		out:        out,
+		sigintChan: make(chan os.Signal, 1),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "dump-config POD ...",
+		Short: "dump proxy config",
+		Long:  dumpConfigDescription,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			dumpConfigCmd.pod = args[0]
+			conf, err := config.RESTClientGetter.ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig")
+			}
+			dumpConfigCmd.config = conf
+
+			clientset, err := kubernetes.NewForConfig(conf)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster. Check kubeconfig")
+			}
+			dumpConfigCmd.clientSet = clientset
+			return dumpConfigCmd.run()
+		},
+		Example: dumpConfigExample,
+	}
+
+	//add mesh name flag
+	f := cmd.Flags()
+	f.StringVarP(&dumpConfigCmd.namespace, "namespace", "n", metav1.NamespaceDefault, "Namespace of pod")
+	f.Uint16VarP(&dumpConfigCmd.localPort, "local-port", "p", constants.EnvoyAdminPort, "Local port to use for port forwarding")
+
+	return cmd
+}
+
+func (cmd *proxyDumpConfigCmd) run() error {
+	// Check if the pod belongs to the mesh
+	pod, err := cmd.clientSet.CoreV1().Pods(cmd.namespace).Get(context.TODO(), cmd.pod, metav1.GetOptions{})
+	if err != nil {
+		return errors.Errorf("Could not find pod %s in namespace %s", cmd.pod, cmd.namespace)
+	}
+	if !isMeshedPod(*pod) {
+		return errors.Errorf("Pod %s in namespace %s is not a part of a mesh", cmd.pod, cmd.namespace)
+	}
+
+	portForwarder, err := NewPortForwarder(cmd.config, cmd.clientSet, cmd.pod, cmd.namespace, cmd.localPort, constants.EnvoyAdminPort)
+	if err != nil {
+		return errors.Errorf("Error setting up port forwarding: %s", err)
+	}
+
+	err = portForwarder.Start(func(pf *PortForwarder) error {
+		url := fmt.Sprintf("http://localhost:%d/config_dump", cmd.localPort)
+
+		// #nosec G107: Potential HTTP request made with variable url
+		resp, err := http.Get(url)
+		if err != nil {
+			return errors.Errorf("Error fetching url %s: %s", url, err)
+		}
+		config, err := ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
+		if err != nil {
+			return errors.Errorf("Error retrieving proxy config: %s", err)
+		}
+		fmt.Fprintf(cmd.out, "%s", config)
```

Comment:
There's almost no chance this really matters, but could we stream `resp.Body` directly to `cmd.out` with `io.Copy` instead of buffering the whole thing with `ioutil.ReadAll`? That may be a bit more readable aside from any marginal performance gains.

Reply:
Will give this a try.